hbase-shell/src/main/ruby/shell.rb

```diff
@@ -320,10 +323,20 @@ def eval_io(io)
       scanner = RubyLex.new
       scanner.set_input(io)
 
+      scanner.each_top_level_statement do |statement, linenum|
+        puts(workspace.evaluate(nil, statement, filename, linenum))
+      end
+      nil
+    end
+
+    ##
+    # Runs a block and logs exception from both Ruby and Java, optionally discarding the traceback
+    #
+    # @param [Boolean] hide_traceback if true, Exceptions will be converted to
+    #   a SystemExit so that the traceback is not printed
+    def self.exception_handler(hide_traceback)
       begin
-        scanner.each_top_level_statement do |statement, linenum|
-          puts(workspace.evaluate(nil, statement, 'stdin', linenum))
-        end
+        yield
```

Comment:
Whats going on here? We are skipping output of the exception stack trace?

Reply:
We need this begin/rescue w/ a yield in the middle of it?

Reply:
The `yield` ensures that whatever block is passed to `exception_handler` is executed within our begin/rescue, which is how we are able to catch and translate exceptions to a SystemExit to avoid printing the full traceback.